### DISCUSSION TITLE
Status of the App - Help needed 🙏

### DISCUSSION THREAD

**Discussion by mrzmyr** [mrzmyr/pixy-mood-tracker-app#202] [Announcements]
This is not so easy to write because I have to admit something: there is currently a bug in the app that prevents me from updating the app. But that's also the idea of open source: to get stuck and ask for help.

So here are the details…

- The app is built with [Expo](expo.dev), which allows it to push updates without natively building it and going through app reviews. I used this from Version 1.68.0 (which is online in the App Store right now) to Version 1.81.0 (which is the version within the app itself right now)
- After I created the native build for 1.82.0 and published it to the beta testers, some were reporting that they lost their data
- When opening the app after the update, the calendar view only shows a spinner (even after closing and opening the app again)
- When exporting the [`items`](https://github.com/mrzmyr/pixy-mood-tracker-app/blob/v1.82.0/src/hooks/useLogs.tsx#L162) key is empty `[]` but the [`tags`](https://github.com/mrzmyr/pixy-mood-tracker-app/blob/v1.82.0/src/hooks/useTags.tsx#L145) and [`settings`](https://github.com/mrzmyr/pixy-mood-tracker-app/blob/v1.82.0/src/hooks/useSettings.tsx#L108) still hold the correct data

**Assumptions**
- It seems like it has nothing to do with the native build itself as it resets the `items` to an empty array `[]` and not the whole storage (excluding the possibility that the underlying data has been corrupted or deleted).

**Possible entry points for investigation**
- The data is loaded in `useLogs` on the first render using [`load`](https://github.com/mrzmyr/pixy-mood-tracker-app/blob/v1.82.0/src/hooks/useLogs.tsx#L162)
- When [`import`](https://github.com/mrzmyr/pixy-mood-tracker-app/blob/v1.82.0/src/hooks/useLogs.tsx#L81)ing the data is [`migrate`](https://github.com/mrzmyr/pixy-mood-tracker-app/blob/v1.82.0/src/hooks/useLogs.tsx#L82C23-L82C23)d
- The loading indicator is shown, which means the `loaded` key seems not to be set and taken from the [`INITIAL_STATE`](https://github.com/mrzmyr/pixy-mood-tracker-app/blob/v1.82.0/src/hooks/useLogs.tsx#L153) (see [calendar/index.tsx](https://github.com/mrzmyr/pixy-mood-tracker-app/blob/v1.82.0/src/screens/Calendar/index.tsx#L57) and [`useLogs`](https://github.com/mrzmyr/pixy-mood-tracker-app/blob/v1.82.0/src/hooks/useLogs.tsx#L84))

As I'm short on time currently, I do not have the capacity to look into more details on this. But if you have the time, please let me know, and I can guide you through the source code. Also, if you have any good questions, hints, or directions on how to tackle this, let me know, and I will try to dig deeper.

It warms my heart every single week when I hear what Pixy does for you. Seriously, it means a lot to me. 
Thanks for your continuous support!

**Comment by KR-Law** [comment]
I show that you removed some logging did you get this figured out?

**Reply by mrzmyr** [reply]
> I show that you removed some logging did you get this figured out?

Nope, not yet. Did you have a look?

**Reply by KR-Law** [reply]
Yes I did. Time required for me personally would be too high to deconstruct the issue. 

Have you determined that if this happens only with calendar views where multiple entries in a single day are present?
There is something up with those some beta testers that had the issue that is being triggered.

**Comment by bnjbvr** [comment]
Hi! if this is still an issue, here are a few random thoughts. Maybe they are obvious and won't be any useful, but, just in case this helps for debugging...

- Could the storage key have changed for logs, between the two versions?
- Could some format in the database have changed, leading to entries being dismissed as invalid?
- Did a dependency involved in the database change in the process?
- Has there been a data migration between 1.81 and 1.82 that could explain the loss of data?

(I'd be happy to investigate most of those things myself, but I've never done any Android/iOS dev before :sweat_smile:)

**Comment by PaxJaromeMalues** [comment]
Ist Pixy by this point still being actively developed or has the issue with the data loss damaged the project beyond repair?

**Reply by mrzmyr** [reply]
tbh, I currently do not have the time to investigate more. I'm happy to jump on a fork that tries to fix this if needed. :)
